### ISSUE TITLE
SA1118 Does Not Allow Collection Initalisers

### ISSUE THREAD

**Issue by connelhooley** [DotNetAnalyzers/StyleCopAnalyzers#2916]
I am using the `StyleCop.Analyzers` NuGet package version `1.0.2`.

The following is flagged as an error:

``` csharp
var x = new SomeCustomClass(
    "value1",
    "value2",
    1,
    2,
    new[] // This breaks SA1118
    {
        new SomeOtherCustomClass("item1"),
        new SomeOtherCustomClass("item2"),
        new SomeOtherCustomClass("item3"),
        new SomeOtherCustomClass("item4"),
        new SomeOtherCustomClass("item5"),
    },
    5);
```

Given StyleCop allows anonymous objects to span multiple lines due to indentation (https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/2650), it should also allow collections. I'm not sure if this is a bug or a missing feature.

**Comment by bluetentacle** [comment]
Seconded! Don't understand why object initializers are allowed but not array initializers

**Comment by bddckr** [comment]
I'm hitting the same issue. Trying to figure out what needs to be done here. Is it just following [this commit](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/commit/9b7ea58fbfbd35577ff982dffb3652c10243b2f1) and basically just adding `SyntaxKind.ArrayCreationExpression` to `ArgumentExceptionSyntaxKinds`?

**Comment by SeijiSuenaga** [comment]
I agree that this is pretty much equivalent to the case addressed by the commit that @bddckr linked.

**Comment by Lonli-Lokli** [comment]
Is there any update?
